ext/lz4ruby/lz4ruby.c

```diff
@@ -126,6 +126,90 @@ static VALUE lz4internal_compressHC(VALUE self, VALUE header, VALUE input, VALUE
   return compress_internal(LZ4_compressHC, header, input, in_size);
 }
 
+static VALUE lz4internal_compress_with_dict(VALUE self,
+                                            VALUE header,
+                                            VALUE input,
+                                            VALUE in_size,
+                                            VALUE dictionary,
+                                            VALUE dictSize) {
+  const char *src_p;
+  int src_size;
+
+  const char *header_p;
+  int header_size;
+
+  VALUE result;
+  char *buf;
+  int buf_size;
+
+  int comp_size;
+
+  LZ4_stream_t* const lz4Stream = LZ4_createStream();
+  LZ4_resetStream(lz4Stream);
```

Comment:
Why do we need to reset the stream straight after creating it? Is this a quirk of LZ4?